metakernel/magics/shell_magic.py

```diff
@@ -47,11 +46,10 @@ def line_shell(self, *args):
         if os.path.exists(cwd):
             os.chdir(cwd)
 
-        if resp:
-            self.kernel.Print(resp)
-
-    def eval(self, cmd):
-        return self.repl.run_command(cmd, stream_handler=self.kernel.Print)
+    def eval(self, cmd, incremental=False):
+        stream_handler = self.kernel.Print if incremental else None
+        return self.repl.run_command(cmd, timeout=None,
```

Comment:
It is only ignored when we pass `incremental=True` (where the results are handled by line for us) or we don't care about the output.